src/algebra/module/basic.lean

```diff
@@ -253,6 +254,14 @@ library_note "vector space definition"
 abbreviation vector_space (R : Type u) (M : Type v) [field R] [add_comm_group M] :=
 semimodule R M
 
+/-- A semimodule over a semiring automatically inherits a `mul_action_with_zero` structure. -/
+@[priority 100] -- see Note [lower instance priority]
+instance semimodule.to_mul_action_with_zero [semiring R] [add_comm_monoid M] [semimodule R M] :
+  mul_action_with_zero R M :=
+{ smul_zero := smul_zero,
+  zero_smul := zero_smul R,
+  ..(infer_instance : mul_action R M) }
+
```

Comment:
You should put this all the way up on line 61, and then remove the lemma on line 63

Reply:
If I remove 
```lean
@[simp] theorem zero_smul : (0 : R) • x = 0 := semimodule.zero_smul x
```

all the proofs that used `simp` to call `zero_smul` do not work...

Reply:
Yes, that's what https://github.com/leanprover-community/mathlib/pull/6590#discussion_r593103251 is for